server/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -547,6 +557,13 @@ private SearchContext findContext(long id, TransportRequest request) throws Sear
     }
 
     final SearchContext createAndPutContext(ShardSearchRequest request) throws IOException {
+        if (numActiveContexts >= MAX_OPEN_CONTEXT.get(settings)) {
```

Comment:
Can this function be called by multiple threads at the same time?  And if so, should `numActiveContexts` be volatile?

Reply:
Yes it can @mayya-sharipova but we restrict the counting to `scroll` queries let's use an AtomicInteger. The overhead should be limited since we create the search context for a scroll query only on the initial request.